tests/ctest/test_highlevel.c

```diff
@@ -0,0 +1,1145 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include <elektra.h>
+#include <elektra_conversion.h>
+#include <tests_internal.h>
+
+static const char * parentKeyName = "user/tests/highlevel";
+static char * test_name = "";
+static int test_line = 0;
+
+static void fatalErrorHandler (ElektraError * error)
+{
+	nbError++;
+	printf ("%s:%d: fatal error in test %s: %s \n", __FILE__, test_line, test_name, elektraErrorDescription (error));
+	ElektraKDBError * kdbError = elektraErrorLowLevelError (error);
+	if (kdbError != NULL)
+	{
+		const char * severityString;
+		switch (elektraKDBErrorSeverity (kdbError))
+		{
+		case ELEKTRA_ERROR_SEVERITY_ERROR:
+			severityString = "ERROR";
+			break;
+		case ELEKTRA_ERROR_SEVERITY_WARNING:
+			severityString = "WARNING";
+			break;
+		default:
+		case ELEKTRA_ERROR_SEVERITY_FATAL:
+			severityString = "FATAL";
+			break;
+		}
+
+		printf ("\tKDB %s %d [%s/%s]: %s\n", severityString, elektraKDBErrorCode (kdbError), elektraKDBErrorGroup (kdbError),
+			elektraKDBErrorModule (kdbError), elektraKDBErrorDescription (kdbError));
+		printf ("\t\tReason: %s\n", elektraKDBErrorReason (kdbError));
+		int warningCount = elektraKDBErrorWarningCount (kdbError);
+		printf ("\t\t%d Warnings:\n", warningCount);
+		const ElektraKDBError ** warnings = elektraKDBErrorWarnings (kdbError);
+		for (int i = 0; i < warningCount; ++i)
+		{
+			printf ("\t\t - Warning %d [%s/%s]: %s\n", elektraKDBErrorCode (warnings[i]), elektraKDBErrorGroup (warnings[i]),
+				elektraKDBErrorModule (warnings[i]), elektraKDBErrorDescription (warnings[i]));
+		}
+		printf ("\t\tFrom Key: %s\n", keyName (elektraKDBErrorKey (kdbError)));
+	}
+
+	yield_error ("fatal");
+	ElektraErrorCode code = elektraErrorCode (error);
+	elektraFree (error);
+	exit (code);
+}
+
+#define SETUP_TEST(NAME)                                                                                                                   \
+	test_line = __LINE__ - 2;                                                                                                          \
+	test_name = "test_" #NAME;                                                                                                         \
+	resetKDB ();
+
+static Elektra * createElektra (KeySet * defaults, ElektraError ** error)
+{
+	Elektra * elektra = elektraOpen ("user/tests/highlevel", defaults, error);
+
+	if (error == NULL)
```

Comment:
Technically, because of the way the function is called `elektra != NULL` and `error == NULL` are equivalent here. But you are right, checking `elektra` would be more reliable.